### PR TITLE
prov/verbs: Add NIC information into fi_info

### PR DESCRIPTION
**Pull request by dmitrygladkov**
Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

**Comment by dmitrygladkov** [comment]
@nikitaxgusev please, take a look

File: prov/verbs/src/verbs_info.c [hunk 1]
```diff
@@ -645,6 +769,17 @@ static int fi_ibv_alloc_info(struct ibv_context *ctx, struct fi_info **info,
 	fi->tx_attr->caps = ep_dom->caps;
 	fi->rx_attr->caps = ep_dom->caps;
 
+	if (!(fi->nic = ofi_nic_dup(NULL))) {
```
Comment (shefty) [inline prov/verbs/src/verbs_info.c]: Please avoid assignments inside if statements.  Some compilers still complain about this.

**Comment by a-ilango** [comment]
fi_info failure when provider is built with dl option:
```
libfabric:core:core:ofi_ini_dir():413<warn> dlopen(.../ofi_rhel7-master-dl/17/install/lib/libfabric/libverbs-fi.so): .../ofi_rhel7-master-dl/17/install/lib/libfabric/libverbs-fi.so: undefined symbol: ofi_nic_dup
```

**Comment by dmitrygladkov** [comment]
@a-ilango @shefty  should we use `fi_control(..., FI_DUP)` from `default_nic_ops` instead?
@jsquyres does usNIC provider work when it is built with dl option?

**Comment by shefty** [comment]
Coming in through the main API is usually ideal when that's a possibility.  Is that usable here?

**Comment by shefty** [comment]
Do we just need to move ofi_nic_dup() into common.c?

**Comment by jsquyres** [comment]
@dmitrygladkov It used to.  I just tested master head right now and it doesn't seem to work.  😦 

Any idea what broke?

**Comment by shefty** [comment]
The introduction of common functions and usnic using them 'broke' the dl build.  The functions and structure all need to move to common.c, or some other shared file.

**Comment by jsquyres** [comment]
Super sad.  😦  What common functions did we use that we shouldn't have?

**Comment by shefty** [comment]
I'm pretty sure they should have been used, just moved into a shared file.

**Comment by jsquyres** [comment]
Ah, I think you're referring to, for example, 8d97e7164a, where we call things like `ofi_nic_dup()`.  That's currently in `src/fabric.c`, and `ofi_nic_dup()` is not part of `libfabric.map`, and is therefore not exported from libfabric.  If `ofi_nic_dup()` moved to `src/common.c`, it would be visible to the providers, even in DL situations.

Right?

**Comment by shefty** [comment]
@jsquyres - yes.

**Comment by jsquyres** [comment]
@dmitrygladkov I confirm that it works for usnic now.  Thanks!

**Comment by jsquyres** [comment]
What's up with Intel CI failing?  There's no "Details" link to see what is going wrong.

**Comment by aingerson** [comment]
@jsquyres node configuration error. I've triggered a rebuild. Should be happy soon.

**Comment by shefty** [comment]
@jsquyres - our CI is behind the firewall and we can't figure out a way to expose the details without exposing some internal network details.

**Comment by dmitrygladkov** [comment]
@jsquyres @shefty should we include the 1st commit to v1.7.x?

**Comment by dmitrygladkov** [comment]
@a-ilango @aingerson Please, take a look at the 3rd commit

**Comment by shefty** [comment]
@dmitrygladkov - yes, please add the 1st commit to 1.7.x  Thanks

**Pull request merged**